app/src/main/java/eu/kanade/domain/manga/model/Manga.kt

```diff
@@ -99,6 +99,38 @@ data class Manga(
         it.initialized = initialized
     }
 
+    fun copyFrom(other: SManga): Manga {
+        var manga = this
+        if (other.author != null) {
+            manga = manga.copy(author = other.author)
+        }
+
+        if (other.artist != null) {
+            manga = manga.copy(artist = other.artist)
+        }
+
+        if (other.description != null) {
+            manga = manga.copy(description = other.description)
+        }
+
+        if (other.genre != null) {
+            manga = manga.copy(genre = other.getGenres())
+        }
+
+        if (other.thumbnail_url != null) {
+            manga = manga.copy(thumbnailUrl = other.thumbnail_url)
+        }
+
+        manga = manga.copy(status = other.status.toLong())
+
+        manga = manga.copy(updateStrategy = other.update_strategy)
+
+        if (!initialized) {
+            manga = manga.copy(initialized = other.initialized)
+        }
+        return manga
```

Comment:
```suggestion
        val author = other.author ?: author
        val artist = other.artist ?: artist
        val description = other.description ?: description
        val genres =  if (other.genre != null) {
            other.getGenres()
        } else {
            genres
        }
        val thumbnailUrl = other.thumbnail_url ?: thumbnailUrl
        return manga.copy(
            author = author,
            artist = artist,
            description = description,
            genres = genres,
            thumbnailUrl = thumbnailUrl,
            status = other.status.toLong(),
            updateStrategy = other.update_strategy,
            initialized = other.initialized && initialized
        )
```
Do we really need to create so many copies of this? I think this is more efficient.